### ISSUE TITLE
Support for editing SQLite over ssh

### ISSUE THREAD

**Issue by hpohlmeyer** [TablePlus/TablePlus#2952]
I would like to see support for editing SQLite databases over ssh. Some companies are currently looking for good ways the run databases on the edge and SQLite seems to be a popular choice (LiteFS, Chiselstrike Turso …), so editing SQLite over ssh will be more popular in the future.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
SQLite

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
Version 5.3.0 (486)

3. The steps to reproduce this issue:
	
	1. Connect to a remote SQLite database
	2. Try to change a record
	3. You will see:
	<img width="260" alt="image" src="https://user-images.githubusercontent.com/6564077/219577007-39bf3a83-d38f-4501-8c8f-fbf66d025b3e.png">

**Comment by jay-sync** [comment]
same here, I wonder if there is a way to help in speeding up this

**Comment by tors** [comment]
+1 !

**Comment by tsukumijima** [comment]
+1

**Comment by digitalmio** [comment]
+1 for libsql support

**Comment by sadovojav** [comment]
+1

**Comment by jasongitmail** [comment]
This might be a separate issue, but +1 for support for Turso's Libsql, whose [client libraries](https://github.com/tursodatabase/libsql#getting-started) support an HTTPS interface (in addition to a local SQLite file).

**Comment by jorisvdz** [comment]
+1

**Comment by ericcch** [comment]
+1 For gui editing. For those who are looking for a quick workaround, open SQL query tab and write the query instead. This is what I'm doing now.

**Comment by nbbaier** [comment]
Also +1 this! Would love to use TablePlus with Turso!

**Comment by adidoes** [comment]
+1 for this 👆

**Comment by moooss** [comment]
+1

**Comment by jasongitmail** [comment]
In the meantime, [Dataflare](https://github.com/DataflareApp/Dataflare) which supports Libsql. I'm using it with Turso.

**Comment by tomblixt** [comment]
+1 for libsql support

**Comment by kaleabmelkie** [comment]
+1 for libsql support

**Comment by Xexr** [comment]
+1 for libsql/turso support

**Comment by jorisvdz** [comment]
+1

**Comment by glommer** [comment]
Turso founder here - and creator of the libSQL project. Just want the tableplus developers to know that if there's anything we can do to help make that happen, please let us know!

**Comment by Xexr** [comment]
Great news @huyphams. Will it be coming to all client platforms?

**Comment by huyphams** [comment]
@Xexr yes, all platforms macOS, iOS, Windows, Linux and also Bolt SQL https//boltsql.com
<img width="1289" alt="Screenshot 2024-03-14 at 5 17 33 PM" src="https://github.com/TablePlus/TablePlus/assets/3654622/2c090902-74c6-420c-986d-7a8d3d24d8ec">

**Comment by glommer** [comment]
@huyphams please tag @tursodatabase on Twitter when you release. We'll tweet about it until our fingers fall.

**Comment by huyphams** [comment]
Sure thanks @glommer

**Comment by huyphams** [comment]
Hi guys, if you have a chance you can download the build that support turso here: https://www.dropbox.com/scl/fi/jqjjqi6y4qlua49l07wb8/TablePlus.dmg?rlkey=kvh9my6jx330ydj464edyyuad&dl=0

Please let me know if you find any bug, so we can make sure it will be addressed before the official release.
Thanks.

**Comment by housser** [comment]
> Hi guys, if you have a chance you can download the build that support turso here: https://www.dropbox.com/scl/fi/jqjjqi6y4qlua49l07wb8/TablePlus.dmg?rlkey=kvh9my6jx330ydj464edyyuad&dl=0
> 
> Please let me know if you find any bug, so we can make sure it will be addressed before the official release. Thanks.

I've just downloaded and smooth connection to my Turso database.

I did have one error when trying to load a specific table:

> The operation couldn’t be completed. (LibSQL.QuantumValue.QuantumError error 0.)


If it's of any help, as perhaps it's to do with a specific column, the Drizzle-defined schema for this table is:
```
export const position = sqliteTable("position", {
  id: integer("id").primaryKey(),
  boat_id: integer("boat_id").references(() => boat.id),
  latitude: real("latitude"),
  longitude: real("longitude"),
  cog: real("cog"),
  sog: real("sog"),
  heading: real("heading"),
  horizontal_accuracy: real("horizontal_accuracy"),
  mmsi: integer("mmsi"),
  uuid: text("uuid"),
  connection_type: text("connection_type"),
  battery_level: real("battery_level"),
  time_utc: text("time_utc"),
  timestamp: integer("timestamp", { mode: "timestamp_ms" }),
  raw: text("raw"),
  user_agent: text("user_agent"),
},
  (table) => {
    return {
      boat_id_index: index("position_boat_id_idx").on(table.boat_id),
      mmsi_index: index("position_mmsi_idx").on(table.mmsi),
      time_utc_index: index("position_time_utc_idx").on(table.time_utc),
      timestamp_index: index("position_timestamp_idx").on(table.timestamp),
    };
  },
);
```

Happy to provide any other details if I can be helpful, as this is incredibly helpful to me!

**Comment by huyphams** [comment]
Thanks @housser TablePlus does not support defined data type. Let me take a look at it and add to the driver.

**Comment by huyphams** [comment]
It's supported loading blob now @housser you can try to download the patch again (using the same dropbox link above).
Thanks.

**Comment by housser** [comment]
@huyphams I've got the latest version, and now when I try to load that specific table I get an Error popup, but it appears to be a JSON response, with 300 rows of data:

```
[
  {
    "results": {
      "columns": [
        "id",
        "boat_id",
        "latitude",
        "longitude",
        "cog",
        "sog",
        "heading",
        "horizontal_accuracy",
        "mmsi",
        "uuid",
        "time_utc",
        "timestamp",
        "raw",
        "user_agent",
        "connection_type",
        "battery_level"
      ],
      "rows": […]
    }
  }
]
```

Other tables load and interact without issues, so presumably something to do with one of these column types (`REAL`? That's the only type I'm not using in any other tables)

**Comment by huyphams** [comment]
Fixed it @housser you can try again.

**Comment by suabahasa** [comment]
> Nothing @glommer everything is good, thank you so much. We will release the new update with libSQL support in the next release.

will it also include the SQLite over SSH?

**Comment by huyphams** [comment]
It is not possible to implement it at the moment @suabahasa

**Comment by mwiles** [comment]
@huyphams I noticed the DropBox URL was for the Mac Client?  Any ETA on the Linux client release?  We are evaluating Turso and I am unable to connect (running Arch Linux using AUR TablePlus version) while my colleague on TablePlus Mac have this functionality available.  My current version is: 0.1.252-1.

**Comment by huyphams** [comment]
Hi @mwiles it is only available on macOS, Windows and iOS at the moment. We will bring it to Linux later.

**Comment by apphancer** [comment]
Unable to use this feature as I am getting a strange behaviour. Using TablePlus Version 6.0.4 (556) on macOS.
TablePlus connects to server but no tables are listed. 

Interestingly, when the connection is first created, TablePlus connects to it without errors, but no tables are listed.
In Connections.plist, there's the following entry for this connection 

```
<key>DatabasePath</key>
<string>/Users/m/Library/Application Support/com.tinyapp.TablePlus/Cache/Data</string>
```

The local db file does indeed exist at this path `/Users/m/Library/Application Support/com.tinyapp.TablePlus/Cache/Data/data.db`, but it is obviously empty.

However, when editing the connection, the relevant part in Connections.plist changes as follows:

```
<key>DatabasePath</key>
<string>/Users/m/Library/Application Support/com.tinyapp.TablePlus/Cache/Data/4099FDFD-95EF-4BBC-A41B-43C34AD49FB8/data.db</string>
```

This file does not exist. This specific issue might be related to https://github.com/TablePlus/TablePlus/issues/2568

However, the main issue is the empty DB in the first place.

**Comment by LukasPolak** [comment]
> > will it also include the SQLite over SSH?
>
> It is not possible to implement it at the moment suabahasa

@huyphams can we expect SQLite over SSH anytime soon? 🙏

**Comment by anthonywebb** [comment]
+1 for full SQLite over SSH, surprised you cant edit DB over ssh?

**Comment by huyphams** [comment]
Please vote and don't comment, thanks!
